app/controllers/application_controller.rb

```diff
@@ -1,3 +1,9 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  before_action :set_locale
+
+  def set_locale
+    I18n.locale = current_user.try(:locale) || I18n.default_locale
```

Comment:
for the try, you can use this syntax `current_user.&locale` which is equivalent to the try